internal/types/document_validation_test.go

```diff
@@ -102,12 +103,24 @@ func TestDocumentValidateData(t *testing.T) {
 			)),
 			reason: errors.New(`invalid value: { "bar": [ "baz", [ "qaz" ] ] } (nested arrays are not supported)`),
 		},
+		"InfFromUpdate": {
+			doc:      must.NotFail(NewDocument("v", math.Inf(1))),
+			isUpdate: true,
+			reason: errors.New(`update produces invalid value: { "v": +Inf } ` +
+				`(update operations that produce infinity values are not allowed)`),
+		},
+		"NegativeZeroFromUpdate": {
+			doc:      must.NotFail(NewDocument("v", math.Copysign(0, -1))),
+			isUpdate: true,
+			reason: errors.New(`update produces invalid value: { "v": -0 } ` +
```

Comment:
I guess, it's impossible to return just `0` in this case, right? Do we want to differentiate `-0` and `0` when we do operations? 
In diff.md it says "update operation producing -0 is converted to 0.", then why do we expect an error here?

Reply:
Or, as I wrote in my other comment, this validation doesn't have to be in `types`, and then we don't have this problem at all.

Reply:
We now change -0 to 0 in validation.